### DISCUSSION TITLE
Sim2Real understanding using PyTorch

### DISCUSSION THREAD

**Discussion by dhruvkm2402** [Toni-SM/skrl#82] [Sim2Real]
Hi @Toni-SM 
I have a trained PyTorch model for a mobile robot using SKRL module. I want to now deploy it on the actual robot using ROS1. I have gone through the scripts of the real-world for FRANKA robot. One question I had was that the eval script uses torch to load the model. The task script does not have pytorch based implementation. Is there a wrapper you use to convert numpy to pytorch for using it as inputs to the trained agent?

**Comment by Toni-SM** [comment]
Hi @dhruvkm2402 

In the simplest implementation, torch is required to load the agent checkpoint (which, apart from the state dictionary of the models, may contain the state dictionary of the input preprocessor, if one was defined during training).

For an example using ROS/ROS2, you can visit the [KUKA LBR iiwa real-world example](https://skrl.readthedocs.io/en/latest/intro/examples.html#real-world-examples) in **skrl**'s docs.


If you define your environment to inherit from `gym.Env` or `gymnasium.Env` (as show in the KUKA LBR iiwa example), it is not necessary to perform the numpy/torch or torch/numpy conversion within the environment implementation (which uses ROS). **skrl**, automatically takes care of such conversion when the environment is wrapped.

https://github.com/Toni-SM/skrl/blob/6b8b70fc2f5fd13087c70a9b51f0ef630c638bcc/docs/source/examples/real_world/kuka_lbr_iiwa/reaching_iiwa_real_ros_env.py#L12
